### PR TITLE
cudf jit parser now supports .pragma instructions with quotes

### PR DESCRIPTION
**Pull request by robertmaynard**
## Description
During testing of CUDA 12.3 it was found that python user defined functions could generate .pragma instructions that have quotes, so as "nounroll" and "used_bytes_mask N".

The jit parser now makes sure to escape quotes that are part of a `.pragma` instruction so that it can be properly passed to nvrtc/jitify as a CUDA header.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/tests/CMakeLists.txt [hunk 1]
```diff
@@ -617,6 +617,11 @@ ConfigureTest(
 # * bin tests ----------------------------------------------------------------------------------
 ConfigureTest(LABEL_BINS_TEST labeling/label_bins_tests.cpp)
 
+# ##################################################################################################
+# * jit tests ----------------------------------------------------------------------------------
+ConfigureTest(JIT_PARSER_TEST jit/parse_ptx_function.cpp)
+target_include_directories(JIT_PARSER_TEST PRIVATE "$<BUILD_INTERFACE:${CUDF_SOURCE_DIR}/src>")
```
Comment (vyasr) [inline cpp/tests/CMakeLists.txt:623]: What do we need this include for? What header is getting pulled?

**Review by vyasr — APPROVED** [review]
One question about the CMake, otherwise looks good. I'm curious, how did you generate the tests? I assume you didn't write the PTX from scratch. I'm wondering if it would be helpful to include a compilation line in the test file demonstrating how to generate that PTX to make it easier to add future tests if needed.

**Review by PointKernel — APPROVED** [review]

File: cpp/tests/CMakeLists.txt (hunk 1 above)
Comment (robertmaynard) [reply cpp/tests/CMakeLists.txt:623]: The `<jit/parser.hpp>` file includes other headers that are private to cudf. So I needed this include to test the JIT parser in isolation.

**Comment by robertmaynard** [comment]
> One question about the CMake, otherwise looks good. I'm curious, how did you generate the tests? I assume you didn't write the PTX from scratch. I'm wondering if it would be helpful to include a compilation line in the test file demonstrating how to generate that PTX to make it easier to add future tests if needed.


I generated the PTX by dumping existing PTX code from libcudf with `cuobjdump`. From that I selected subsection of existing functions that had relevant bits to test ( differerent pragmas ) and merged it with an empty function signature. Since we only need to parse the PTX I wasn't worried about the correctness of the function, only that it contained valid instructions.

**Pull request merged**